### PR TITLE
Add new feature: SHOW DATABASES like 'xxx'

### PR DESCRIPTION
**Pull request by hexiaoting**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add new feature: SHOW DATABASES LIKE 'xxx'

Detailed description / Documentation draft:
Support:
```
SHOW DATABASES LIKE 'XXX';
SHOW DATABASES NOT LIKE 'XXX';
SHOW DATABASES LIKE 'XXX' LIMIT XXX;
```
relate to issue: https://github.com/ClickHouse/ClickHouse/issues/14518

### CODE REVIEW

File: src/Interpreters/InterpreterShowTablesQuery.cpp [hunk 1]
```diff
@@ -31,7 +31,20 @@ String InterpreterShowTablesQuery::getRewrittenQuery()
...
     /// SHOW DATABASES
     if (query.databases)
-        return "SELECT name FROM system.databases";
+    {
+        std::stringstream rewritten_query;
+        rewritten_query << "SELECT name FROM system.databases";
+
+        if (!query.like.empty())
+        {
+            rewritten_query << " WHERE name " << (query.not_like ? "NOT " : "") << "LIKE " << std::quoted(query.like, '\'');
```
Comment (alexey-milovidov) [inline src/Interpreters/InterpreterShowTablesQuery.cpp:40]: Looks like ILIKE would not work?
The same question for SHOW TABLES.

File: src/Parsers/ASTShowTablesQuery.cpp [hunk 2]
```diff
@@ -18,6 +18,20 @@ void ASTShowTablesQuery::formatQueryImpl(const FormatSettings & settings, Format
     if (databases)
     {
         settings.ostr << (settings.hilite ? hilite_keyword : "") << "SHOW DATABASES" << (settings.hilite ? hilite_none : "");
+
+        if (!like.empty())
```
Comment (alexey-milovidov) [inline src/Parsers/ASTShowTablesQuery.cpp:22]: Let's separate LIKE and LIMIT printing to functions to avoid copy-paste.

**Comment by hexiaoting** [comment]
@alexey-milovidov 
I have added ILIKE for "SHOW DATABASES/TABLES". 


But is there  a bug with ilike  ???
It is not work with system.databases

```
dell123 :) SELECT name FROM system.databases                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
┌─name─────────│
| default         │ 
│ sqlgraph      │ 
│ system         │  
└─────────────┘   

dell123 :) SELECT name FROM system.databases WHERE name ILIKE 'De%'
OK.
0 rows in set.
```

But it work well when query in normal tables
```
dell123 :) select * from test;
┌─i─┬──────────j─┬─name───────┐
│ 1 │          2 │ hewenting2 │
│ 1 │          4 │ aa         │
│ 2 │          3 │            │
│ 4 │          3 │Heaa            │
└───┴────────────┴────────────┘

dell123 :) select * from test where name ilike 'He%';
┌─i─┬─j─┬─name───────┐
│ 1 │ 2 │ hewenting2 │
│ 4 │ 3 │ Heaa │
└───┴───┴──────┘
```
If so , I want to try to fix it.

**Comment by myrrc** [comment]
> @alexey-milovidov
> I have added ILIKE for "SHOW DATABASES/TABLES".
> 
> But is there a bug with ilike ???
> It is not work with system.databases
> 
> ```
> dell123 :) SELECT name FROM system.databases                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
> ┌─name─────────│
> | default         │ 
> │ sqlgraph      │ 
> │ system         │  
> └─────────────┘   
> 
> dell123 :) SELECT name FROM system.databases WHERE name ILIKE 'De%'
> OK.
> 0 rows in set.
> ```
> 
> But it work well when query in normal tables
> 
> ```
> dell123 :) select * from test;
> ┌─i─┬──────────j─┬─name───────┐
> │ 1 │          2 │ hewenting2 │
> │ 1 │          4 │ aa         │
> │ 2 │          3 │            │
> │ 4 │          3 │Heaa            │
> └───┴────────────┴────────────┘
> 
> dell123 :) select * from test where name ilike 'He%';
> ┌─i─┬─j─┬─name───────┐
> │ 1 │ 2 │ hewenting2 │
> │ 4 │ 3 │ Heaa │
> └───┴───┴──────┘
> ```
> 
> If so , I want to try to fix it.

The ILIKE operator works only in SELECT context as for now, I'll make the PR for SHOW TABLES/DATABASES and mention it here.

**Comment by hexiaoting** [comment]
@myrrc  OK^^

**Comment by alexey-milovidov** [comment]
Style check has found tabs in code (need to replace to spaces).

**Comment by hexiaoting** [comment]
> Style check has found tabs in code (need to replace to spaces).

I have fixed it .

**Comment by alexey-milovidov** [comment]
@alesapin What's about the `01465_ttl_recompression` test?

**Comment by alesapin** [comment]
> @alesapin What's about the `01465_ttl_recompression` test?

flaky test.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@hexiaoting I don't remember what has changed but now it's Ok:
```
milovidov-desktop :) SELECT name FROM system.databases WHERE name ILIKE 'De%'

SELECT name
FROM system.databases
WHERE name ILIKE 'De%'

┌─name────┐
│ default │
└─────────┘
```

**Pull request merged**